src/doc/performance-tuning.md

```diff
@@ -110,95 +205,184 @@ Node 1 HugePages_Total:    32
+receiving packets can evict other entries in L3 cache.  If your ring
+buffer is too big, it can evict other data resident in cache that you
+might want.
+
+Another down-side of having a big buffer is latency.  The bigger your
+buffer, the more the bloat.  Usually, Snabb applications always run
+faster than the incoming packets, so the buffer size isn't an issue when
+everything is going well; but in a traffic spike where packets come in
+faster than Snabb can process them, the buffer can remain full the whole
+time, which just adds latency for the packets that Snabb does manage to
+process.
+
+However, too small a buffer exposes the user to a higher risk of
+packet loss due to jitter in the Snabb breath time.  A "breath" is one
+cycle of processing a batch of packets, and as we mentioned it is
+currently up to 128 packets at a time.  This processing doesn't always
+take the same amount of time: the contents of the packets can
+obviously be different, causing different control flow and different
+data flow.  Even well-tuned applications can exhibit jitter in their
+breath times due to differences between what data is in cache (and
+which cache) and what data has to be fetched from memory.  Infrequent
+events like reconfiguration can also cause one breath to take longer
+than another.  Poorly-tuned applications might have other sources of
+latency such as garbage collection, though this is not usually the
+case in production-ready Snabb network functions like the NFV virtual
+switch or the lwAFTR.
+
+So, a bigger ring buffer insulates packet processing from breath
+jitter.  You want your ring buffer to be big enough to not drop
+packets due to jitter during normal operation, but not bigger than
+that.  In our testing we usually use the default ring buffer size.  In
+your operations you might want to increase this up to 2048 entries.
+We have not found that bigger ring buffer sizes are beneficial, but it
+depends very much on the environment.
+
+## Virtualization
+
+Running network functions under a Snabb NFV virtual switch is a great
+for isolation, however you do have to get a number of things right or
+you are not going to have a good experience.  Really we should link
+out to another guide here, but until that happens, here are some brief
+notes.
+
+### Host-side switches: Snabb NFV, Snabb vMX
+
+All of the above considerations apply to running a virtual switch --
+give each virtual switch instance a dedicated CPU core on the host,
+bind it to the same NUMA node of the card and the CPU, and make sure
+there are enough huge pages.
+
+Additionally, for good performance, the host switch must run on the
+same NUMA node as the guest.
+
+### QEMU & Vhost-User
+
+[Vhost-User](http://www.virtualopensystems.com/en/solutions/guides/snabbswitch-qemu/)
+is used to connect Snabb with a high performance virtual interface
+attached to a QEMU-based virtual machine.  This requires hugepages
+made available to Qemu:
 
 ```
-$ lspci|grep 10-
```

Comment:
Hmm. Probably just "it was in the previous doc" bias. README.running.md and README.troubleshooting.md mention it, and as you said it's basic, so I think that's good enough.